src/rdkafka_partition.c

```diff
@@ -186,6 +186,21 @@ static void rd_kafka_toppar_consumer_lag_tmr_cb (rd_kafka_timers_t *rkts,
 	rd_kafka_toppar_consumer_lag_req(rktp);
 }
 
+/**
+ * Enqueue an RD_KAFKA_OP_BARRIER type of operation
+ * when the op_version is updated.
+ *
+ * Locality: toppar handler thread
+ */
+void rd_kafka_toppar_op_version_bump (rd_kafka_q_t *rktp_fetchq,
+		                             int32_t version) {
+        rd_kafka_op_t *rko;
+
```

Comment:
I think you should move the actual bumping of the op version into this function as well:
```
rktp->rktp_op_version = version;
```